client/src/pages/SettingsPage/SettingsPage.module.scss

```
@@ -0,0 +1,111 @@
+        display: flex;
+        flex-direction: column;
+        justify-content: center;
+        align-items: center;
+        background-color: #f5f5f5;
+        border-radius: 10px;
+        padding-left: 20px;
+        padding-right: 20px;
+        box-shadow: 0 0 10px rgba(0, 0, 0, 0.1);
+        height: calc(100vh - 150px);
+        @media (max-width: 768px) { // For tablet and mobile
+            max-height: calc(100vh - 340px); // Used chatGPT to calculate the height more accurately
+        }
+        &__menu{
+            @media (max-width: 768px) { // For tablet and mobile
+                max-height: 30vh;
+                overflow-y: auto;
+            }
+        }
+    }
+
+    &__two_column_grid_item__menu {
+        display: flex;
+        flex-direction: column;
+        justify-content: flex-start;
+        align-items: center;
+        background-color: #f5f5f5;
+        border-radius: 10px;
+        padding-left: 20px;
+        padding-right: 20px;
+        box-shadow: 0 0 10px rgba(0, 0, 0, 0.1);
+        @media (max-width: 768px) { // For tablet and mobile
+            max-height: 30vh;
+            padding-left: 20px;
+        }
+    }
+
+    &__grid_container {
+        padding-left: 20px;
+        padding-right: 20px;
+    }
+
+    &__menu_container {
+        display: flex;
+        flex-direction: column;
+        align-items: center;
+        justify-content: center;
+
+        // Start of code generated by ChatGPT
+        @media (max-width: 768px) {
+            flex-direction: row; // Lays menu items out in a row on smaller screens
+            justify-content: center; // This will center the items in the container
+            align-items: center;
+            flex-wrap: wrap; // Allows items to wrap into multiple rows if needed
+            max-height: auto;
+        }
+        // End of ChatGPT generated code
+    }
+
+    &__menu_item {
+        display: flex;
+        flex-direction: column;
+        justify-content: center;
+        text-align: center;
+        align-items: center;
+        border-bottom: 1px solid ;
+        border-bottom: 1px solid #ccc;
+
+        @media (max-width: 768px) {
+            border-bottom: none;
+            border-right: 1px solid #ccc;
+            padding-right: 10px;
+            padding-left: 10px;
+            &:last-child {
+                border-right: none; // Remove right border for the last item
+            }
+        }
+    }
+    
+    &__component_container {
+        display: flex;
+        flex-direction: column;
+        width: 100%;
+        height: 100%;
+        padding-left: 20px;
+        padding-right: 20px;
+        padding-top: 10px;
+        justify-content: flex-start;
+        align-items: stretch;
+        border-radius: 10px;
+        box-shadow: 0 0 10px rgba(0, 0, 0, 0.1);
+        max-height: calc(100vh - 250px);
+        overflow-y: auto;
+
+        @media (max-width: 768px) {
+            width: 100%;
+        }
```

Once again, just converting the `px` values to `rem`. Feel free to adjust the values to your liking.
```suggestion
        height: 100%;
        padding-left: 1.25rem;
        padding-right: 1.25rem;
        padding-top: 0.75rem;
        justify-content: flex-start;
        align-items: stretch;
        border-radius: 0.75rem;
        box-shadow: 0 0 0.75rem rgba(0, 0, 0, 0.1);
        max-height: calc(100vh - 15.75rem);
        overflow-y: auto;

        @media (width <= 48.00rem) {
            width: 100%;
        }
```